src/main.c

```diff
@@ -3440,17 +3440,36 @@ int main(int argc, char **argv)
         conf.error_log = NULL;
     }
 
-    { /* raise RLIMIT_NOFILE */
-        struct rlimit limit;
+    {
+        struct rlimit limit = {0};
+
         if (getrlimit(RLIMIT_NOFILE, &limit) == 0) {
+            if (conf.max_connections > (int)limit.rlim_max) {
+                fprintf(stderr, "[error] The 'max-connections'=[%d] configuration value "
+                                "should not exceed the file descriptor limit of "
+                                "the process 'RLIMIT_NOFILE'=[%lu]\n",
+                        conf.max_connections, limit.rlim_max);
+                return EX_CONFIG;
+            }
+
+            /* raise RLIMIT_NOFILE */
```

Comment:
Nothing more than a nitpick: I get the idea of moving the comment here, though it's a bit awkward to start this block without a comment and then have a comment describing what is done at the middle of the block. How about moving the comment back to where is was, changing it to something like, "raise RLIMIT_NOFILE, making sure that we can reach max_connections"?